ui/selectors/multichain.ts

```diff
@@ -212,9 +212,36 @@ export function getMultichainNetwork(
   // on having a non-EVM account being selected!
   const selectedAccount = account ?? getSelectedInternalAccount(state);
   const nonEvmNetworks = getMultichainNetworkProviders(state);
-  const nonEvmNetwork = nonEvmNetworks.find((provider) => {
-    return provider.isAddressCompatible(selectedAccount.address);
-  });
+  const selectedChainId = state.metamask.selectedMultichainNetworkChainId;
+
+  let nonEvmNetwork: MultichainProviderConfig | undefined;
+
+  // First try to find network by selectedChainId
+  if (selectedChainId) {
+    nonEvmNetwork = nonEvmNetworks.find(
+      (provider) => provider.chainId === selectedChainId,
+    );
+  }
+
+  // If no network found by selectedChainId, we try to find by scopes
+  if (!nonEvmNetwork && selectedAccount.scopes?.length > 0) {
+    // If we have a selectedChainId but didn't find a match, we try to find a network
+    // that matches both the selectedChainId and is in the scopes
+    if (selectedChainId) {
+      nonEvmNetwork = nonEvmNetworks.find(
+        (provider) =>
+          provider.chainId === selectedChainId &&
+          selectedAccount.scopes.includes(provider.chainId),
+      );
+    }
+  }
+
+  // If still no network found, we try to find a network that is address compatible
+  if (!nonEvmNetwork) {
+    nonEvmNetwork = nonEvmNetworks.find((provider) => {
+      return provider.isAddressCompatible(selectedAccount.address);
+    });
```

Comment:
I think it will be time to get rid of this logic. The `isAddressCompatible` is not something that is supported by the networks from the `MultichainNetworkController`, so we should start removing the "hardcoded providers" at some point.

So maybe leaving a `TODO` to remove this would be great (to not scope creep your PR)

```suggestion
    // TODO: This runtime logic is not supported by the `MultichainNetworkController`, we should remove this and rely
    // only on the network configs provided by this controller.
    nonEvmNetwork = nonEvmNetworks.find((provider) => {
      return provider.isAddressCompatible(selectedAccount.address);
    });
```